### PR TITLE
DBZ-5229 define centralized and modular aproach for debezium storage

### PR DESCRIPTION
**Pull request by blcksrx**

### CODE REVIEW

File: debezium-connector-mysql/pom.xml [hunk 1]
```diff
@@ -115,6 +115,12 @@
             <artifactId>kafka_${version.kafka.scala}</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.debezium</groupId>
+            <artifactId>debezium-storage-kafka</artifactId>
+            <version>2.0.0-SNAPSHOT</version>
```
Comment (jpechane) [inline debezium-connector-mysql/pom.xml:121]: The module definition should go to `debezium-bom/pom.xml` so version is maintained there - see `Debezium artifacts`. Also please add the dependency right below `debezium-core`

File: debezium-connector-sqlserver/pom.xml [hunk 2]
```diff
@@ -88,6 +88,12 @@
             <artifactId>kafka-connect-avro-converter</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (jpechane) [inline debezium-connector-sqlserver/pom.xml:91]: Same as for mysql

File: debezium-storage/debezium-storage-kafka/pom.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            <artifactId>slf4j-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.kafka</groupId>
+            <artifactId>connect-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (jpechane) [inline debezium-storage/debezium-storage-kafka/pom.xml:33]: Please add missing newline.

File: debezium-storage/pom.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>debezium-storage</artifactId>
+    <name>Debezium Storage Module</name>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>debezium-storage-kafka</module>
+    </modules>
+</project>
```
Comment (jpechane) [inline debezium-storage/pom.xml:17]: Please add missing newline.

File: debezium-storage/debezium-storage-kafka/src/main/java/io/debezium/storage/kafka/KafkaDatabaseHistory.java [hunk 5]
```diff
@@ -3,7 +3,7 @@
  *
  * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
  */
-package io.debezium.relational.history;
+package io.debezium.storage.kafka;
```
Comment (jpechane) [inline debezium-storage/debezium-storage-kafka/src/main/java/io/debezium/storage/kafka/KafkaDatabaseHistory.java:6]: Should be `io.debezium.storage.history.kafka`

**Review by jpechane — COMMENTED** [review]
@blcksrx This is definitely good start! I left couple of comments. Could you please also handle Oracle module as I think it uses the storage too.

After that it would be nice if you could introduce file storage module and move `FileDatabaseHistory` there.

**Comment by blcksrx** [comment]
@jpechane Thank you very much. I did the changes.
There are some questions left:
1. How we want to handle the default value of `database.history`? [Since `KafkaDatabaseHistory now is in the separate package]
2. How we want to handle `database.history` related test? [shall we test each connector compatibility on the storages module or  reverse?

File: debezium-connector-mysql/pom.xml [hunk 6]
```diff
@@ -115,7 +123,7 @@
             <artifactId>kafka_${version.kafka.scala}</artifactId>
             <scope>test</scope>
         </dependency>
-    </dependencies>
+           </dependencies>
```
Comment (jpechane) [inline debezium-connector-mysql/pom.xml:126]: Incorrect indentation?

**Review by jpechane — COMMENTED** [review]
@blcksrx Nice work!
I think thesre is only one indentation issue left and documentation/modules/ROOT/pages/operations/logging.adoc also contains the old package names. Otherwise we are good to go!

**Comment by blcksrx** [comment]
@jpechane 
Thank you for notes,
could you please answer my questions above?

**Comment by jpechane** [comment]
@blcksrx

1. Please introduce dependency on `debezium-storage-kafka` to connectors as it should be provided by default in the installaton package.
2. I am not sure what exactly do you mean - could you please elaborate more - point to a test example?

**Comment by blcksrx** [comment]
@jpechane 
Thank you very much.
About second note. Im gonna mention this [KafkaDatabaseHistoryTest](https://github.com/debezium/debezium/blob/main/debezium-connector-mysql/src/test/java/io/debezium/relational/history/KafkaDatabaseHistoryTest.java) as an example in the `mysql-connector`

**Comment by blcksrx** [comment]
@jpechane Do you need smth to do on my side about this PR?

**Comment by jpechane** [comment]
@blcksrx Applied thanks a lot!

**Comment by blcksrx** [comment]
Thanks. Im going to work on S3 Storage

**Pull request merged**